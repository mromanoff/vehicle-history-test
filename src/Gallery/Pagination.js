import React from 'react';
import PropTypes from 'prop-types';
import Pagination from 'react-js-pagination';
import styled from 'styled-components';

const StyledPagination = styled.div`
  margin-bottom: 2rem;
  > ul {
    display: flex;
    justify-content: center;
    margin: 0;
    padding: 0;

    > li {
      margin-right: 0.5rem;
      border-radius: 50%;
      background-color: ${props => props.theme.color.grayLight};
      font-size: ${props => props.theme.size.sm};
      color: ${props => props.theme.color.white};
      transition: background-color 100ms ease-out 100ms;
      display: flex;
      align-items: center;
      justify-content: center;
      text-align: center;
      width: 46px;
      height: 46px;
      font-weight: 300;

      &:last-child {
        margin-right: 0;
      }

      &.active {
        color: ${props => props.theme.color.white};
        background-color: ${props => props.theme.color.primary5};
      }

      &:hover {
        background-color: ${props => props.theme.color.primary5};
      }

      &.disabled {
        display: none;
      }

      > a,
      span {
        color: ${props => props.theme.color.white};
        text-decoration: none;
      }
    }
  }
`;

const GalleryPagination = ({
  currentPage,
  itemsCountPerPage,
  totalItemsCount,
  onChange,
}) => {
  const pageCount = Math.ceil(totalItemsCount / itemsCountPerPage);

  return (
    pageCount > 1 && (
      <StyledPagination>
        <Pagination
          activePage={currentPage}
          itemsCountPerPage={itemsCountPerPage}
          totalItemsCount={totalItemsCount}
          pageRangeDisplayed={3}
          onChange={onChange}
        />
      </StyledPagination>
    )
  );
};

export default GalleryPagination;

GalleryPagination.propTypes = {
  currentPage: PropTypes.number.isRequired,
  itemsCountPerPage: PropTypes.number.isRequired,
  totalItemsCount: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
};
