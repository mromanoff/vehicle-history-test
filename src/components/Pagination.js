import React from 'react';
import PropTypes from 'prop-types';
import Pagination from 'react-js-pagination';

const ProviderPagination = ({
  page,
  itemsCountPerPage,
  totalItemsCount,
  onChange,
}) => {
  const pageCount = Math.ceil(totalItemsCount / itemsCountPerPage);

  return (
    pageCount > 1 && (
      <div className="module">
        <Pagination
          innerClass="pagination"
          activePage={page}
          itemsCountPerPage={itemsCountPerPage}
          totalItemsCount={totalItemsCount}
          pageRangeDisplayed={3}
          onChange={onChange}
        />
      </div>
    )
  );
};

export default ProviderPagination;

ProviderPagination.propTypes = {
  page: PropTypes.number.isRequired,
  itemsCountPerPage: PropTypes.number.isRequired,
  totalItemsCount: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
};
