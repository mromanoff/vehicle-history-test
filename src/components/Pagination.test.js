import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from './Pagination';

it('Pagination renders correctly', () => {

  let props = {
    page: 1,
    itemsCountPerPage: 25,
    totalItemsCount: 100,
    onChange: () => {},
  };

  const tree = renderer.create(<Pagination {...props} />).toJSON();
  expect(tree).toMatchSnapshot();

  // // manually trigger the callback
   props.totalItemsCount = 0;
    // re-rendering
  // tree = component.toJSON();
  //expect(tree).toMatchSnapshot();

});
