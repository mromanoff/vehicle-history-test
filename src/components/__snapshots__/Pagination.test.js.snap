// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<div
  className="module"
>
  <ul
    className="pagination"
  >
    <li
      className="disabled"
      onClick={[Function]}
    >
      <a
        className=""
        href="#"
      >
        «
      </a>
    </li>
    <li
      className="disabled"
      onClick={[Function]}
    >
      <a
        className=""
        href="#"
      >
        ⟨
      </a>
    </li>
    <li
      className="active"
      onClick={[Function]}
    >
      <a
        className="undefined"
        href="#"
      >
        1
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <a
        className=""
        href="#"
      >
        2
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <a
        className=""
        href="#"
      >
        3
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <a
        className=""
        href="#"
      >
        ⟩
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <a
        className=""
        href="#"
      >
        »
      </a>
    </li>
  </ul>
</div>
`;
